src/apply_rules.rs

```diff
@@ -73,6 +75,102 @@ struct PropertyRule {
     property_value: String,
 }
 
+#[derive(Clone, Default, Debug, Deserialize)]
+struct NetworkConsolidation {
+    #[serde(flatten)]
+    network: Network,
+    grouped_from: Vec<String>,
+}
+
+fn read_networks_consolidation_file<P: AsRef<Path>>(
+    networks_consolidation_file: P,
+) -> Result<Vec<NetworkConsolidation>> {
+    info!("Reading networks consolidation rules.");
+
+    #[derive(Debug, Deserialize)]
+    struct Consolidation {
+        #[serde(rename = "networks")]
+        networks_consolidation: Vec<NetworkConsolidation>,
+    }
+
+    let file = File::open(networks_consolidation_file)?;
+    let reader = BufReader::new(file);
```

Comment:
Won't we affect the performance by doing this? I can see something in the documentation about using `BufReader` might help for short reads from a file
https://docs.serde.rs/serde_json/fn.from_reader.html

> When reading from a source against which short reads are not efficient, such as a `File`, you will want to apply your own buffering because `serde_json` will not buffer the input. See `std::io::BufReader`.